data-mongodb/src/main/java/io/micronaut/data/mongodb/serde/DataDecoderContext.java

```diff
@@ -181,11 +182,17 @@ public <T> Deserializer<? extends T> findDeserializer(Argument<? extends T> type
             return mappedCodec.deserializer;
         }
         if (codec != null) {
-            // Eliminate codecs for basic types and collections
-            Class<? extends T> encoderClass = codec.getEncoderClass();
-            if (!ClassUtils.isJavaLangType(encoderClass)
-                && !Map.class.isAssignableFrom(encoderClass)
-                && !Iterable.class.isAssignableFrom(encoderClass)) {
+            // Eliminate codecs for basic types (except byte array) and collections
+            boolean useCodec;
```

Comment:
Can you please extract the check into some util class?

Reply:
Makes sense, done in b46944aaf433ca513592de8607c709b6a77ccab9